app/src/main/java/org/mozilla/fenix/experiments/NimbusFeatures.kt

```diff
@@ -0,0 +1,120 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.experiments
+
+import android.content.Context
+import org.mozilla.experiments.nimbus.mapKeysAsEnums
+import org.mozilla.fenix.Config
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.getVariables
+
+/**
+ * Component for exposing nimbus Feature Variables.
+ * For more information see https://experimenter.info/feature-variables-and-me
+ *
+ * @param context - A [Context] for accessing the feature variables from nimbus.
+ */
+class NimbusFeatures(private val context: Context) {
+
+    val homeScreen: HomeScreenFeatures by lazy {
+        HomeScreenFeatures(context)
+    }
+
+    /**
+     * Component that indicates which features should be active on the home screen.
+     */
+    class HomeScreenFeatures(private val context: Context) {
+        /**
+         * `FeatureId.HOME_PAGE` feature; the complete JSON, is shown here:
+         *
+         * ```json
+         * {
+         *     "sections-enabled": {
+         *         "topSites": true,
+         *         "recentlySaved": false,
+         *         "jumpBackIn": false,
+         *         "pocket": false,
+         *         "recentExplorations": false
+         *     }
+         * }
+         * ```
+         */
+
+        /**
+         * This enum accompanies the `FeatureId.HOME_PAGE` feature.
+         *
+         * These names here should match the names of entries in the JSON.
+         */
+        @Suppress("EnumNaming")
+        private enum class HomeScreenSection(val default: Boolean) {
+            topSites(true),
+            recentlySaved(false),
+            jumpBackIn(false),
+            pocket(false),
+            recentExplorations(false);
+
+            companion object {
+                /**
+                 * CreateS a map with the corresponding default values for each sections.
+                 */
+                fun toMap(context: Context): Map<HomeScreenSection, Boolean> {
+                    return values().associate { section ->
+                        val channelDefault = if (section == pocket) {
+                            FeatureFlags.isPocketRecommendationsFeatureEnabled(context)
+                        } else {
+                            Config.channel.isNightlyOrDebug
+                        }
+                        section to (channelDefault || section.default)
+                    }
+                }
+            }
+        }
+
+        private val homeScreenFeatures: Map<HomeScreenSection, Boolean> by lazy {
+            val experiments = context.components.analytics.experiments
+            val variables = experiments.getVariables(FeatureId.HOME_PAGE, false)
+            val sections: Map<HomeScreenSection, Boolean> =
+                variables.getBoolMap("sections-enabled")?.mapKeysAsEnums()
+                    ?: HomeScreenSection.toMap(context)
```

Comment:
I changed the initial implementation a bit as we were returning an empty map `mapOf()` when the variables where not available.  See the original implementation from your patch below:

```kotlin
    private val homescreenSectionVisibility: Map<SectionId, Boolean> by lazy {
        val experiments = appContext.components.analytics.experiments
        val variables = experiments.getVariables(FeatureId.HOME_PAGE, false)
        val values: Map<SectionId, Boolean> = variables.getBoolMap("sections-enabled")?.mapKeysAsEnums() ?: mapOf()
        values.withDefault { key -> key.default }
    }
```

It was causing that we don't have any entries when we don't have variables, that will be the case of nightly at the moment.

Reply:
This is the only thing left. 

`FeatureFlags` and these flags from Nimbus shouldn't mix. If you want to default everything on for nightly

```kotlin
    private val homescreenSectionVisibility: Map<SectionId, Boolean> by lazy {
        val experiments = appContext.components.analytics.experiments
        val variables = experiments.getVariables(FeatureId.HOME_PAGE, false)
        val values: Map<SectionId, Boolean> = variables.getBoolMap("sections-enabled")?.mapKeysAsEnums() ?: mapOf()
        values.withDefault { key -> key.default || Config.channel == ReleaseChannel.Nightly }
    }
```

For debug, you should use [`initial_experiments.json`][3] (You need to increase the count to 10000).

[3]: https://github.com/mozilla-mobile/fenix/pull/21411/files#diff-cfdb9388f7e08573f26394a641d2e28d18b862b11553fd5b3e7c30edd1837ccfR70

Reply:
`values.withDefault { key -> key.default || Config.channel == ReleaseChannel.Nightly }`

This line doesn't get executed on nightly, as we don't have the nimbus variables there and we are falling back to `mapOf()` causing that `values.withDefault` doesn't get executed as there are no keys.

Reply:
any call to  any feature ( e.g `homeScreenFeatures[HomeScreenSection.jumpBackIn] == true`) will return `false`, causing the we don't show any feature on nightly.

Reply:
Please do a manual a test and let me know if the features are showing on nightly variant.

Reply:
This is video demo of using the suggested approach on nightly.

```
        private val homeScreenFeatures: Map<HomeScreenSection, Boolean> by lazy {
            val experiments = context.components.analytics.experiments
            val variables = experiments.getVariables(FeatureId.HOME_PAGE, false)
            val values: Map<HomeScreenSection, Boolean> = variables.getBoolMap("sections-enabled")?.mapKeysAsEnums() ?: mapOf()
            values.withDefault { key -> key.default || Config.channel == ReleaseChannel.Nightly }
        }
```
  

https://user-images.githubusercontent.com/773158/135653798-30f357de-5cfe-4463-acf0-ea4c81cf09fd.mp4